lib/ipinfo-rails/ipselector/xforwarded_ipselector.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+require 'ipinfo-rails/ipselector/ipselector_interface'
+
+class XForwardedIPSelector
+    include IPSelectorInterface
+    def initialize(request)
+        @request = request
+    end
+    
+    def get_ip()
+        return @request.env['HTTP_X_FORWARDED_FOR']
```

Comment:
Not sure if Rails does some of its own manipulations, but we want an algorithm matching this for example: https://github.com/ipinfo/laravel/blob/master/src/iphandler/OriginatingIPSelector.php